artifact-management/commands/gem-test.yml

```diff
@@ -1,44 +1,87 @@
     type: env_var_name
     default: AVVO_GEM_REPO_TOKEN
+  bundler_1_version:
+    description: The Bundler version that should be used for "bundle install" and the actual test command
+    type: string
+    default: 1.17.3
+  bundler_2_version:
+    description: The Bundler version that should be used for "bundle install" and the actual test command
+    type: string
+    default: 2.1.4
   gem_test_dir:
-    description: The directory in which the gem tesets should run
+    description: The directory in which the gem tests should run
     type: string
     default: .
-  packagecloud_read_token:
-    description: The token to be used when fetching packages from Packagecloud.
-    type: env_var_name
-    default: PACKAGECLOUD_READ_TOKEN
   test_command:
-    description: The command that should be used to run tests
+     default: bundle exec rake test
+     description: The command that should be used to run tests
+     type: string
+  ruby_versions:
+    description: The ruby versions to be used to test the gem.
     type: string
-    default: bundle exec rake test
+    default: ""
 steps:
   - setup-gem-environment
-  - run:
-      name: Install bundler gem
-      command: |-
-        cd <<parameters.gem_test_dir>>
-        gem install bundler --version=<<parameters.bundler_version>>
-  - run:
-      name: bundle install
-      command: |-
-        cd <<parameters.gem_test_dir>>
-        bundle install
-  - run:
-      name: Run tests
-      command: |-
-        cd <<parameters.gem_test_dir>>
-        <<parameters.test_command>>
+  - when:
+      condition: <<parameters.ruby_versions>>
+      steps:
+        - run:
+            name: Install bundler gem
+            command: |-
+              for v in $(echo "<<parameters.ruby_versions>>" | sed "s/,/ /g");
+              do \
+                if [[ "$v" > "2.6.9999" ]];
+                then
+                    echo "Installing bundler version 2 for Ruby version $v"
+                    rvm $v do gem install bundler --version=<<parameters.bundler_2_version>>
+                else
+                    echo "Installing bundler version 1 for Ruby version $v"
+                    rvm $v do gem install bundler --version=<<parameters.bundler_1_version>>
+                fi \
+              done
+        - run:
+            name: bundle install
+            command: |-
+              cd <<parameters.gem_test_dir>>
+              rvm <<parameters.ruby_versions>> do bundle install
+        - run:
+            name: Run tests
+            command: |-
+              cd <<parameters.gem_test_dir>>
+              rvm <<parameters.ruby_versions>> do <<parameters.test_command>>
+  - unless:
+     condition: <<parameters.ruby_versions>>
+     steps:
+       - run:
+           name: Install bundler gem
+           command: |-
+             ACTUAL_RUBY=$(ruby --version);
+             RUBY_VERSION=${ACTUAL_RUBY:5:3};
+             if [[ "$RUBY_VERSION" > "2.6.9999" ]]
```

Comment:
In the link https://tldp.org/LDP/abs/html/comparison-ops.html says:
>
```
is greater than, in ASCII alphabetical order
if [[ "$a" > "$b" ]]
if [ "$a" \> "$b" ]
Note that the ">" needs to be escaped within a [ ] construct.
See Example 27-11 for an application of this comparison operator.
```
Then using [ ] constructor (and escape “>”) to standardize with other gems in same orb (like setup-gem-environment)

Reply:
I think we can use the `[[ ]]` construct (if we require bash - see the previous conversation about this), but I think you're missing the point:

```
if [[ 17 > 3 ]]; then
  echo bigger
else
  echo smaller
fi

smaller
```

**That result is wrong** because you're comparing in alphabetical order. 17 is greater than 3, not less than. You need to require bash and then use `(( ))` instead of `[[ ]]` to get _numeric_ comparison, which is what you are looking for.

Reply:
😳 I apologize about this rude mistake. No excuse. 
New proposed solution should work in any Bourne shell; but a cons about it is that it assumes ruby 1.x is never used. What you think?

Reply:
Yeah, I think this is good enough. We don't use Ruby 1.x anywhere.

Of course, it's possible to write a compound predicate that considers the case where `(major >=3) || (major == 2 && minor >=7)` but I'm not sure it's worth it.

Also, `mayor` in Spanish is spelled `major` in English. But everyone knows what you meant.